src/main/kotlin/com/openlattice/transporter/services/DataTransporterService.kt

```diff
@@ -0,0 +1,137 @@
+import com.google.common.eventbus.Subscribe
+import com.google.common.util.concurrent.ListeningExecutorService
+import com.hazelcast.core.HazelcastInstance
+import com.openlattice.assembler.AssemblerConnectionManager
+import com.openlattice.assembler.AssemblerConnectionManagerDependent
+import com.openlattice.data.storage.partitions.PartitionManager
+import com.openlattice.datastore.services.EdmManager
+import com.openlattice.datastore.services.EntitySetManager
+import com.openlattice.edm.EntitySet
+import com.openlattice.edm.events.*
+import com.openlattice.edm.set.EntitySetFlag
+import com.openlattice.edm.type.EntityType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.transporter.processors.TransporterPropagateDataEntryProcessor
+import com.openlattice.transporter.processors.TransporterSynchronizeTableDefinitionEntryProcessor
+import com.openlattice.transporter.types.TransporterColumnSet
+import com.zaxxer.hikari.HikariDataSource
+import io.prometheus.client.Histogram
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.*
+import java.util.concurrent.Future
+
+@Service
+final class DataTransporterService(
+        private val eventBus: EventBus,
+        private val dataModelService: EdmManager,
+        private val partitionManager: PartitionManager,
+        private val entitySetService: EntitySetManager,
+        private val executor: ListeningExecutorService,
+        hazelcastInstance: HazelcastInstance
+): AssemblerConnectionManagerDependent<Void?>
+{
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(DataTransporterService::class.java)
+        val pollTimer: Histogram = Histogram.build()
+                .namespace("transporter")
+                .name("poll_duration_seconds")
+                .help("time to do one database sync poll")
+                .register()
+    }
+
+    private lateinit var acm: AssemblerConnectionManager
+    private lateinit var transporter: HikariDataSource
+
+    private val transporterState = HazelcastMap.TRANSPORTER_DB_COLUMNS.getMap( hazelcastInstance )
+
+    private fun syncTable(et: EntityType): Optional<Future<*>> {
+        val prev = transporterState.putIfAbsent(et.id, TransporterColumnSet(emptyMap()))
+        return if (prev == null) {
+            val props = dataModelService.getPropertyTypes(et.properties)
+            Optional.of(
+                    transporterState.submitToKey(et.id, TransporterSynchronizeTableDefinitionEntryProcessor(props).init(acm))
+            )
+        } else {
+            Optional.empty()
+        }
+    }
+
+    override fun init(acm: AssemblerConnectionManager): Void? {
+        this.acm = acm
+        this.transporter = acm.connect("transporter")
+        executor.submit {
+            val entityTypes = dataModelService.entityTypes.toList()
+            logger.info("initializing DataTransporterService with {} types", entityTypes.size)
```

Comment:
Just a minor thing, but can you please use uppercase for the log entries first word?